core/src/main/java/org/elasticsearch/search/aggregations/support/AggregationPath.java

```diff
@@ -217,10 +199,23 @@ public double resolveValue(HasAggregations root) {
                 continue;
             }
 
-            // the agg can only be a metrics agg, and a metrics agg must be at the end of the path
             if (i != pathElements.size() - 1) {
-                throw new IllegalArgumentException("Invalid order path [" + this +
- "]. Metrics aggregations cannot have sub-aggregations (at [" + token + ">" + pathElements.get(i + 1) + "]");
+                // try get bucket by key name
+                if(agg instanceof InternalMappedTerms){
```

Comment:
As mentioned in my main comment I would personally prefer if we avoided referencing specific Aggregation classes in the aggregator framework code